src/plugins/foxPage/components/OtherOpportunities/FoxOtherOpportunityPanelRow.tsx

```diff
@@ -77,15 +77,16 @@ export const FoxOtherOpportunityPanelRow: React.FC<FoxOtherOpportunityPanelRowPr
     }
 
     if (earnOpportunity) {
-      const { provider, chainId, contractAddress, assetId, rewardAddress } = earnOpportunity
-      const { assetReference } = fromAssetId(assetId)
+      const { provider, chainId, contractAddress, rewardAddress } = earnOpportunity
       history.push({
         pathname: location.pathname,
         search: qs.stringify({
           provider,
           chainId,
           contractAddress,
-          assetReference,
+          assetReference: earnOpportunity.underlyingAssetId
+            ? fromAssetId(earnOpportunity.underlyingAssetId).assetReference
+            : undefined,
```

Comment:
Maybe undefined in `FoxFarmingEarnOpportunityType` for backwards compatibility reasons - this type is still used as the return type for old foxEthSlice hooks to be removed.

In this particular case, the earn opportunity type for FOX farming *is* actually guaranteed to yield an `underlyingAssetId`